server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3242,14 +3252,17 @@ SnapshotsInProgress computeUpdatedState() {
                     continue;
                 }
                 final List<SnapshotsInProgress.Entry> newEntries = new ArrayList<>(oldEntries.size());
+                // Iterate through the snapshots passing in the updates list. Once an update gets applied to a snapshot, it will be removed
+                // from the updates list passed to the next snapshot.
                 for (SnapshotsInProgress.Entry entry : oldEntries) {
-                    final var newEntry = applyToEntry(entry, updates.getValue());
+                    final var newEntry = applyUpdatesToEntry(entry, updates.getValue());
                     newEntries.add(newEntry);
                     if (newEntry != entry && newEntry.state().completed()) {
                         newlyCompletedEntries.add(newEntry);
                     }
                 }
-                updated = updated.withUpdatedEntriesForRepo(repoName, newEntries);
+                assert updates.getValue().isEmpty();
```

Comment:
I'm curious if this will hold, shall see if testing passes. I think it should be true.

Reply:
Hmm not sure that's true, we might be seeing a retry of an update for a snapshot (maybe one that already completed) and in that case it'd be ignored and left in the list?

In any case I think it'd be nice to have that in a separate PR too. Noisy-but-safe changes like renamings are best kept as separate commits for future bisections.

Reply:
Removed 👍